_posts/2024-06-07-opensearch-performance-2.14.md

```diff
@@ -0,0 +1,161 @@
+
+OpenSearch continues to innovate on the core search engine. Our full plans are available in the open on [GitHub](https://github.com/orgs/opensearch-project/projects/153/views/1). 
+
+We are planning the following enhancements: 
+
+- Supporting fine-grained customer monitoring of query execution, allowing you to identify problematic queries, diagnose query execution bottlenecks and tune query performance.
+- Optimizing common query execution infrastructure by developing multi-level caching (tiered caching); faster interconnections among components using an efficient binary format (Protobuf).
+- Improving query execution itself, using document reordering to optimize query throughput and storage efficiency (graph bipartite partitioning); analyzing and rewriting complex queries to make them run more efficiently; dynamic pruning; count-only caching.
+- Modularizing the query engine to make future improvements faster and easier.
+
+### Hybrid search 
+
+Hybrid search combines lexical and semantic vector search to get the best of both worlds. Highly specific names--- like part numbers---are best found using lexical search, while broader queries are often best handled by semantic vector search. We have supported hybrid search since OpenSearch 2.10. 
+
+We are planning to enhance hybrid search in the following ways:
+
+- By executing the lexical and the vector searches in parallel, we can achieve up to a 25% latency improvement.
+- We will support re-sorting of the hybrid query results.
+- We will support additional algorithms for combining the query results. In particular, reciprocal rank fusion has shown good results in some applications.
+- We will return the raw scores of the subqueries, which are useful for debugging and for relevance tuning.
+
+### Vector search
+
+Semantic vector search is a powerful technique for finding semantic similiarities. However, it can be costly, because indexing vectors to make them efficient for search is time-consuming, and the vectors themselves can consume large amounts of memory.
+
+We are currently working on the following vector search improvements:
+
+- Accelerating vector indexing by 80% and reducing the memory required for indexing.
+- Reducing the amount of space needed to store vectors by compressing vector components from 32 bits to 16, 8, and even 1 bit. Our experiments show modest reductions in search quality (recall) with 16 and 8 bit components; we are still analyzing the 1-bit case (binary quantization).
+- Reducing cost by employing disk-based ANN algorithms that use external storage (SSDs, S3, and so on).
+
+Vector techniques have also been less flexible than lexical techniques in many ways, so we are:
+
+- Improving search relevancy by supporting reranking on compressed vectors.
+- Supporting ANN vector search for the multi-tenant case, where each tenant has its own subcollection. This is especially valuable for customers each of whose user organizations has a large vector collection.
```

Comment:
The last sentence is incorrect as written. Do we just mean something like "This is especially valuable for customers with large vector collections"?

Reply:
Agree